sources/packages/backend/apps/api/src/route-controllers/student-scholastic-standings/_tests_/e2e/student-scholastic-standings.institutions.controller.getScholasticStandingSummary.e2e-spec.ts

```diff
@@ -0,0 +1,163 @@
+    // Arrange
+    const student = await saveFakeStudent(db.dataSource);
+    await saveFakeApplication(db.dataSource, {
+      student,
+    });
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const endpoint = `/institutions/scholastic-standing/summary/student/${student.id}`;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.FORBIDDEN)
+      .expect({
+        statusCode: 403,
+        message: INSTITUTION_STUDENT_DATA_ACCESS_ERROR_MESSAGE,
+        error: "Forbidden",
+      });
+  });
+
+  it("Should get the scholastic standing summary for the provided student including the data retrieved from the sfas system when a BC Public Institution requests it.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(db.dataSource);
+    const firstApplication = createFakeApplication({
+      location: collegeFLocation,
+      student,
+    });
+    const { institution } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    collegeCLocation = createFakeInstitutionLocation({ institution });
+    const secondApplication = createFakeApplication({
+      location: collegeCLocation,
+      student,
+    });
+    await applicationRepo.save([firstApplication, secondApplication]);
+    const firstAppScholasticStanding = createFakeStudentScholasticStanding(
+      { submittedBy: student.user, application: firstApplication },
+      {
+        initialValue: { unsuccessfulWeeks: 15 },
+      },
+    );
+    const secondAppScholasticStanding = createFakeStudentScholasticStanding(
+      { submittedBy: student.user, application: secondApplication },
+      {
+        initialValue: { unsuccessfulWeeks: 3 },
+      },
+    );
+    await scholasticStandingRepo.save([
+      firstAppScholasticStanding,
+      secondAppScholasticStanding,
+    ]);
+    await saveFakeSFASIndividual(db.dataSource, {
+      initialValues: {
+        lastName: student.user.lastName,
+        birthDate: student.birthDate,
+        sin: student.sinValidation.sin,
+        unsuccessfulCompletion: 12,
+      },
+    });
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const endpoint = `/institutions/scholastic-standing/summary/student/${student.id}`;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({ lifetimeUnsuccessfulCompletionWeeks: 30 });
+  });
+
+  it("Should return zero lifetime unsuccessful completion weeks for the provided student as a part of the student scholastic summary when the student has no unsuccessful completion weeks in SIMS or in the SFAS system when a BC Public Institution requests it.", async () => {
```

Comment:
I'm not sure this test is useful. Why testing zero unsuccessful completion weeks is important if we tested the sum of them in the previous test?

Reply:
Zero calculation is kind of like an edge case which is set when the query doesn't return anything. That is why I added it.